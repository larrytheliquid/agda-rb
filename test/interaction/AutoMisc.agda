{-# OPTIONS --universe-polymorphism #-}

module AutoMisc where

-- prelude

postulate
  Level : Set
  lzero : Level
  lsuc  : (i : Level) → Level

{-# BUILTIN LEVEL     Level #-}
{-# BUILTIN LEVELZERO lzero  #-}
{-# BUILTIN LEVELSUC  lsuc   #-}

data _≡_ {a} {A : Set a} (x : A) : A → Set where
  refl : x ≡ x

trans : ∀ {a} {A : Set a} → {x y z : A} → x ≡ y → y ≡ z → x ≡ z
trans refl refl = refl

sym : ∀ {a} {A : Set a} → {x y : A} → x ≡ y → y ≡ x
sym refl = refl

cong : ∀ {a b} {A : Set a} {B : Set b}
       (f : A → B) {x y} → x ≡ y → f x ≡ f y
cong f refl = refl

data _IsRelatedTo_ {a : Level} {Carrier : Set a} (x y : Carrier) : Set a where
  relTo : (x∼y : x ≡ y) → x IsRelatedTo y

begin_ : {a : Level} {Carrier : Set a} → {x y : Carrier} → x IsRelatedTo y → x ≡ y
begin relTo x∼y = x∼y

_∎ : {a : Level} {Carrier : Set a} → (x : Carrier) → x IsRelatedTo x
_∎ _ = relTo refl

_≡⟨_⟩_ : {a : Level} {Carrier : Set a} → (x : Carrier) {y z : Carrier} → x ≡ y → y IsRelatedTo z → x IsRelatedTo z
_ ≡⟨ x∼y ⟩ relTo y∼z = relTo (trans x∼y y∼z)

data ℕ : Set where
  zero : ℕ
  suc  : (n : ℕ) → ℕ

_+_ : ℕ → ℕ → ℕ
zero  + n = n
suc m + n = suc (m + n)

data ⊥ : Set where

¬ : Set → Set
¬ A = A → ⊥

data Π (A : Set) (F : A → Set) : Set where
  fun : ((a : A) → F a) → Π A F

data Σ (A : Set) (F : A → Set) : Set where
  ΣI : (a : A) → (F a) → Σ A F

data Fin : ℕ → Set where
  zero : ∀ {n} → Fin (suc n)
  suc  : ∀ {n} → Fin n → Fin (suc n)

data List (X : Set) : Set where
  []  : List X
  _∷_ : X → List X → List X

_++_ : {X : Set} → List X → List X → List X
[] ++ ys = ys
(x ∷ xs) ++ ys = x ∷ (xs ++ ys)

data Vec (X : Set) : ℕ → Set where
  []  : Vec X zero
  _∷_ : ∀ {n} → X → Vec X n → Vec X (suc n)


module AdditionCommutative where

 lemma : ∀ n m → (n + suc m) ≡ suc (n + m)
 lemma n m = {!!}

 lemma' : ∀ n m → (n + suc m) ≡ suc (n + m)
 lemma' zero m = refl
 lemma' (suc n) m = cong suc (lemma' n m)

 addcommut : ∀ n m → (n + m) ≡ (m + n)
 addcommut n m = {!!}


module Drink where

 postulate RAA : (A : Set) → (¬ A → ⊥) → A

 drink : (A : Set) → (a : A)
            → (Drink : A → Set) → Σ A (λ x → (Drink x) → Π A Drink)
 drink A a Drink = {!!}


module VecMap where

 map : {X Y : Set} → {n : ℕ} → (X → Y) → Vec X n → Vec Y n
 map f xs = {!!} 


module Disproving where

 p : {X : Set} → (xs ys : List X) → (xs ++ ys) ≡ (ys ++ xs)
 p = {!!}
