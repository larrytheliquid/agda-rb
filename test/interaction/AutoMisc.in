top_command (cmd_load currentFile [])
goal_command 0 cmd_auto "-c"
goal_command 1 cmd_auto "-c lemma' -t 60"
goal_command 2 cmd_auto "RAA -t 60"
goal_command 3 cmd_auto "-c"
goal_command 4 cmd_auto "-d Fin"
