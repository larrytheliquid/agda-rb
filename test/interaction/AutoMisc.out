agda2_mode_code (agda2-status-action "")
agda2_mode_code (agda2-status-action "")
agda2_mode_code (agda2-info-action
                 "*All Goals*"
                 "?0 : (n + suc m) ≡ suc (n + m)
?1 : (n + m) ≡ (m + n)
?2 : Σ A (λ x → Drink x → Π A Drink)
?3 : Vec .Y .n
?4 : (xs ys : List .X) → (xs ++ ys) ≡ (ys ++ xs)
")
agda2_mode_code (agda2-goals-action '(0 1 2 3 4))
agda2_mode_code (last . (agda2-make-case-action
                         '("lemma zero m = refl" "lemma (suc n) m = cong suc (lemma n m)")))
agda2_mode_code (agda2-goals-action '(0 1 2 3 4))
agda2_mode_code (last . (agda2-make-case-action
                         '("addcommut zero zero = refl"
                           "addcommut zero (suc n) = sym (cong suc (addcommut n zero))"
                           "addcommut (suc n) m = begin (suc (n + m) ≡⟨ cong suc (addcommut n m) ⟩ (suc (m + n) ≡⟨ sym (lemma' m n) ⟩ ((m + suc n) ∎)))")))
agda2_mode_code (agda2-goals-action '(0 1 2 3 4))
agda2_mode_code (agda2-give-action
                 2
                 "RAA (Σ A (λ x → (x' : Drink x) → Π A Drink))
(λ x →
   x
   (ΣI a
    (λ x' →
       fun
       (λ a' →
          RAA (Drink a')
          (λ x0 →
             x (ΣI a' (λ x1 → fun (λ a0 → RAA (Drink a0) (λ x2 → x0 x1)))))))))")
agda2_mode_code (agda2-status-action "")
agda2_mode_code (agda2-info-action
                 "*All Goals*"
                 "?0 : (n + suc m) ≡ suc (n + m)
?1 : (n + m) ≡ (m + n)
?3 : Vec .Y .n
?4 : (xs ys : List .X) → (xs ++ ys) ≡ (ys ++ xs)
")
agda2_mode_code (agda2-goals-action '(0 1 3 4))
agda2_mode_code (last . (agda2-make-case-action
                         '("map f [] = []" "map f (xs ∷ xs') = f xs ∷ map f xs'")))
agda2_mode_code (agda2-goals-action '(0 1 3 4))
agda2_mode_code (agda2-status-action "")
agda2_mode_code (agda2-info-action
                 "*Auto*"
                 "Listing disproof(s) 0-0
0  (Fin (suc (suc zero))) (suc zero ∷ []) (zero ∷ []) (λ ())
")
agda2_mode_code (agda2-goals-action '(0 1 3 4))
